compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -2532,6 +2547,39 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                         }
                     }
                 }
+                // We special case the situation where `Pin::new` wouldn't work, and isntead
```

Comment:
It's obviously pre-existing, but could you move some of these individual suggestions into their own functions, there's _a lot_ going on in this one function.

Reply:
I'll try to clean this method up in a few days.